File: backend/src/models/QuickAnswer.ts
import {
  Table,
  Column,
  DataType,
  CreatedAt,
  UpdatedAt,
  Model,
  PrimaryKey,
  AutoIncrement,
  ForeignKey,
  BelongsTo
} from "sequelize-typescript";
import Tenant from "./Tenant";

@Table
class QuickAnswer extends Model<QuickAnswer> {
  @PrimaryKey
  @AutoIncrement
  @Column
  id: number;

  @Column(DataType.TEXT)
  shortcut: string;

  @Column(DataType.TEXT)
  message: string;

  @Column(DataType.STRING)
  mediaType: "text" | "buttons" | "list" | "carousel";

  @Column(DataType.TEXT)
  dataJson: string | null;

  @CreatedAt
  createdAt: Date;

  @UpdatedAt
  updatedAt: Date;

  @ForeignKey(() => Tenant)
  @Column(DataType.UUID)
  tenantId: number | string;

  @BelongsTo(() => Tenant)
  tenant: Tenant;
}

export default QuickAnswer;

File: bussines/internal/models/quick_answer.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type QuickAnswer struct {
	ID        int       `gorm:"primaryKey" json:"id"`
	Shortcut  string    `gorm:"not null" json:"shortcut"`
	Message   string    `gorm:"not null" json:"message"`
	TenantID  uuid.UUID `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	MediaType string    `gorm:"column:mediaType" json:"mediaType"`
	DataJson  string    `gorm:"column:dataJson;type:jsonb" json:"dataJson"`
	CreatedAt time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt time.Time `gorm:"column:updatedAt" json:"updatedAt"`
}

func (QuickAnswer) TableName() string {
	return "QuickAnswers"
}
